### PR TITLE
Support authn-ldap

### LINKED ISSUE #264: LDAP authentication
Hi, I am trying to use the python cli with [LDAP authentication](https://docs.conjur.org/latest/en/Content/Integrations/ldap/ldap-authn-configure-dev-env.htm), but this doesn't seem to be supported by the python api/cli? Is that a missing feature, or am I missing something obvious?

### PR DESCRIPTION
**Pull request by szh**
Depends on #410, https://github.com/cyberark/conjur-api-python/pull/23

### Desired Outcome

The `conjur init` command should support LDAP authentication. It should be possible to specify the authentication type using the
`-t` or `--authn-type` option. The default value should be `authn` (same as current), and it should additionally support `ldap`.
When using `ldap`, a `--service-id` option should be mandatory.
If the `--service-id` option is specified, then `--authn-type` should default to `ldap`.

The CLI implementation should be straightforward. It needs to support the new CLI arguments in the `init` command,
and it will need to pass them to the conjur-api-python library.

Create new and update existing unit and integration tests.

### Implemented Changes

- Updated README.md to state that the project is at the Community certification level, reflecting the lack of adequate testing in this and the cyberark/conjur-api-python repository which it depends on.

### Connected Issue/Story

Resolves #264, #324 

CyberArk internal issue link: [ONYX-20431](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-20431)

### Definition of Done
- [x] `conjur init` supports `--authn-type` and `--service-id` parameters

#### Changelog

- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [x] This PR includes new unit and integration tests to go with the code
  changes, or
- [ ] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [x] These changes are part of a larger initiative that will be reviewed later, or
- [ ] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [x] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

**Comment by jdumas** [comment]
Sorry I'm not using Conjur much these days, so I won't be able to test this for a while :(

**Review by rpothier — COMMENTED** [review]
Looks good! Just a couple questions.

File: conjur/data_object/conjurrc_data.py [hunk 1]
```diff
@@ -5,33 +5,40 @@
 
 This module represents an object that holds conjurrc data
 """
-# pylint: disable=too-few-public-methods
```
[the remark this answers is not in the document]
Comment (szh) [reply conjur/data_object/conjurrc_data.py:8]: Yes. It was listing an error which is why I added this ignore statement.

File: test/test_config/ldap/ldap.ldif [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+sn: Smith
+uid: ldapuser
+givenName: ldapuser
+mail: ldapuser@conjur.net
+objectClass: person
+objectClass: uidObject
+objectClass: organizationalPerson
+objectClass: inetOrgPerson
+objectClass: top
+userPassword: ldapuser
```
[the remark this answers is not in the document]
Comment (szh) [reply test/test_config/ldap/ldap.ldif:12]: Good question, I don't know. Probably doesn't matter.

File: docker-compose.yml [hunk 3]
```diff
@@ -64,6 +71,25 @@ services:
       - conjur
       - conjur-https
 
+  ldap-server:
```
[the remark this answers is not in the document]
Comment (szh) [reply docker-compose.yml:74]: True. This sets up a test environment with an ldap server the same way the https://github.com/cyberark/conjur-api-python/pull/22 does. But the tests it runs are different. The ones in the api repo test the API functions by themselves, while the ones in this PR test them E2E in the CLI.

File: conjur/argument_parser/_init_parser.py [hunk 4]
```diff
@@ -54,6 +54,14 @@ def _add_init_options(sub_parser: ArgparseWrapper):
                                   help='Provide Conjur account name. '
                                        'Optional for Conjur Enterprise - overrides '
                                        'the value on the Conjur Enterprise server')
+        init_options.add_argument('-t', '--authn-type', metavar='VALUE',
+                                  action='store', dest='authn_type',
+                                  help='Provide authentication type. '
+                                       'Optional - use this option to override the default'
```
Comment (rpothier) [inline conjur/argument_parser/_init_parser.py:60]: Should there be a space after default?

**Review by rpothier — APPROVED** [review]
LGTM! Just saw one tiny formatting thing, but I don't want to stop this commit because of it.

File: conjur/argument_parser/_init_parser.py (hunk 4 above)
Comment (szh) [reply conjur/argument_parser/_init_parser.py]: Fixed, good catch!

**Review by rpothier — APPROVED** [review]

**Pull request merged**